src/app/modules/account/pages/registration/components/user-registration/user-registration.component.ts

```diff
@@ -28,9 +28,44 @@ import {Account} from "../../../../../../models/account.model";
   templateUrl: "./user-registration.component.html",
   styleUrls: ["./user-registration.component.scss"],
   standalone: true,
-  imports: [IonicModule, CommonModule, FormsModule],
+  imports: [IonicModule, CommonModule, FormsModule,ReactiveFormsModule],
 })
-export class UserRegistrationComponent {
+export class UserRegistrationComponent implements OnInit{
   @Input() account?: Partial<Account>;
-  constructor() {}
+  registrationForm: FormGroup;
+
+  constructor(private fb: FormBuilder) {
+    this.registrationForm = this.fb.group({
+      name: ['', Validators.required],
+      tagline:[''],
+      Description :[''],
+      phoneNumber:['', Validators.pattern(/^-?\d+$/)],
+      alternateEmail:['',[Validators.email]],
+      mailingAddress:[''],
+      emergencyContactNumber:['', Validators.pattern(/^-?\d+$/)],
+      preferredMethodOfContact :[''],
+      occupation:[''],
+      employerName:[''],
+      workExperience:[''],
+      skillsAndExpertise:[''],
+      currentJobTitle:[''],
+      linkedInProfile:[''],
+      educationalBackground:['']
+    });
+  }
+
+  ngOnInit(){
+
+  
+  }
+
+  onSubmit() {
+    if (this.registrationForm.valid) {
+      // Handle form submission
+      const formData = this.registrationForm.value;
+      console.log(formData);
```

Comment:
For this can you look at the code on the `edit-user-profile.ts` file.
```
  onSubmit() {
    if (this.account) {
      // Update the account object with form values
      this.account.email = this.editAccountForm.value.email ?? "";
      this.account.description = this.editAccountForm.value.description ?? "";
      this.account.tagline = this.editAccountForm.value.tagline ?? "";
      this.account.name = this.editAccountForm.value.name ?? "";
      this.account.language = this.editAccountForm.value.language ?? "";

      this.account.userDetails = {
        ...this.account.userDetails,
        firstName: this.account.userDetails?.firstName ?? "",
        lastName: this.account.userDetails?.lastName ?? "",
        username: this.account.userDetails?.username ?? "",
        dateOfBirth: Timestamp.fromDate(
          new Date(this.editAccountForm.value.userDetails?.dateOfBirth ?? ""),
        ),
      };
      this.account.address = {
        ...this.account.address,
        street: this.editAccountForm.value.address?.street ?? "",
        city: this.editAccountForm.value.address?.city ?? "",
        state: this.editAccountForm.value.address?.state ?? "",
        zipcode: this.editAccountForm.value.address?.zipcode ?? "",
        country: this.editAccountForm.value.address?.country ?? "",
        name: this.editAccountForm.value.address?.name ?? "",
        formatted: this.account.address?.formatted ?? "",
        geopoint: this.account.address?.geopoint ?? "",
      };
      this.account.phone = {
        ...this.account.phone,
        number: this.editAccountForm.value.phone?.number ?? "",
        type: this.editAccountForm.value.phone?.type ?? "",
        countryCode: this.editAccountForm.value.phone?.countryCode ?? "",
      };

      this.storeService.updateDoc("accounts", this.account);
    }
  }
  ```
  We need to do something similar to this to save the data for this user.  Then redirect the user to the `user-profile` page.  
`this.router.navigate(["/user-profile/", this.account?.id]);`

Reply:
This code is probably not accurate anymore.  Please checkout the `edit-user-profile.ts` page for the latest version of the update code.